src/binary-reader.cc

```diff
@@ -629,7 +629,7 @@ Result BinaryReader::ReadAddress(Address* out_value,
                                  const char* desc) {
   ERROR_UNLESS(memory < memories.size(),
                "load/store memory %u out of range %lu", memory,
-               memories.size());
+               static_cast<unsigned long>(memories.size()));
```

Comment:
How about changing the format string to "%zu"?

Reply:
Is that 100% portable? I thought it wasn't supported in MSVC.

Reply:
We have at least one other occurrence already so I guess its portable enough:

```
$ git grep "%zu"
src/wat-writer.cc:  WABT_TRACE_ARGS(FlushExprTreeVector, "%zu", expr_trees.size());
```

If it proves non-portable then the correct fix is probably to add a PRI macro like we do for `PRIaddress`

Reply:
According to this discussion MS added support for %z in VS2013: https://stackoverflow.com/questions/15610053/correct-printf-format-specifier-for-size-t-zu-or-iu

Reply:
Done.